### ISSUE TITLE
Add procedure front() and back() to LinkedList standard module

### ISSUE THREAD

**Issue by priyank23** [chapel-lang/chapel#15294]
### Summary of Problem
I think there should be a procedure that returns the first and last element of the LinkedList.

```chapel
use LinkedLists;
var x : LinkedList(int);
x.append(1,2,3);
writeln(x.front());
writeln(x.back());
```
The above code should print 
```
1
3
```

**Comment by bradcray** [comment]
`.first` or `.first()` and `.last` or `.last()` would be more consistent with other Chapel collections, I believe.

**Comment by priyank23** [comment]
Definitely!

**Comment by Aniket21mathur** [comment]
@bradcray @priyank23 I am in favor with the front element, but is there any point in having the last element? (keeping in mind the implementation of linked list)

Thanks!

**Comment by bradcray** [comment]
@Aniket21mathur: From a design perspective you mean?  I think the arguments might be (a) symmetry with other collections, (b) someone curious to know what the last element was, (c) manually iterating through the list and needing to know where to stop (?).

**Comment by Aniket21mathur** [comment]
@bradcray I think I mixed chapel implementation with C++ implementation. I apologise for that. I was thinking in the direction to only have the head pointer of the list and iterate through it till we have a null.

The arguments you proposed are valid, and I agree with them 👍

**Comment by priyank23** [comment]
@bradcray The module also lacks any procedure for searching. Also, can we add a procedure for sorting as well?

**Comment by bradcray** [comment]
No strong opinions here.  Searching seems like an obvious thing to add, and I'd look at the routines in the Search module for inspiration.  I expect sorting a linked list to perform pretty abysmally and don't think that's a common case to worry about (in fact, using linked lists at all in Chapel is probably not a particularly common case).  I.e., it might be "nice to have" but doesn't seem essentially or important to me.  But I'm also not very invested in the linked list module, personally.

**Comment by Liannium** [comment]
I would like to work on adding the first() and last() functions, as well as a search function. Does anyone have any advice on this?

**Comment by priyank23** [comment]
@Liannium  I think using iterative approach would be much better performance wise.

**Comment by bradcray** [comment]
It seems you could write a simple search routine fairly easily by writing something like:

```chapel
for item in myLinkedList do
  if (item == searchForThis) then
    ...
```

**Comment by Liannium** [comment]
Made a PR for font, back, and search. If nobody minds, I would like to implement a sorting algorithm as well.

**Comment by dlongnecke-cray** [comment]
Resolved by #15317 (thanks @Liannium)!
